src/mesh/amr_loadbalance.cpp

```diff
@@ -519,17 +518,6 @@ void Mesh::RedistributeAndRefineMeshBlocks(ParameterInput *pin, ApplicationInput
   int bsf2c = (bnx1 / 2) * ((bnx2 + 1) / 2) * ((bnx3 + 1) / 2) * nx4_tot;
   int bsc2f =
       (bnx1 / 2 + 2) * ((bnx2 + 1) / 2 + 2 * f2) * ((bnx3 + 1) / 2 + 2 * f3) * nx4_tot;
-  // face-centered quantities enrolled in SMR/AMR
-  bssame += num_fc * ((bnx1 + 1) * bnx2 * bnx3 + bnx1 * (bnx2 + f2) * bnx3 +
-                      bnx1 * bnx2 * (bnx3 + f3));
-  bsf2c += num_fc * (((bnx1 / 2) + 1) * ((bnx2 + 1) / 2) * ((bnx3 + 1) / 2) +
-                     (bnx1 / 2) * (((bnx2 + 1) / 2) + f2) * ((bnx3 + 1) / 2) +
-                     (bnx1 / 2) * ((bnx2 + 1) / 2) * (((bnx3 + 1) / 2) + f3));
-  bsc2f +=
-      num_fc *
-      (((bnx1 / 2) + 1 + 2) * ((bnx2 + 1) / 2 + 2 * f2) * ((bnx3 + 1) / 2 + 2 * f3) +
-       (bnx1 / 2 + 2) * (((bnx2 + 1) / 2) + f2 + 2 * f2) * ((bnx3 + 1) / 2 + 2 * f3) +
-       (bnx1 / 2 + 2) * ((bnx2 + 1) / 2 + 2 * f2) * (((bnx3 + 1) / 2) + f3 + 2 * f3));
```

Comment:
I'm not 100% sure where these calculations are going and how they would translate to the new framework. Is it worthwhile to remove this code for now?

Reply:
We can delete this but I think the code above it will need to change. We need to loop over all vars and compute their total size for comms/packing.

Reply:
Agreed. As we talked about in person, the current scheme to compute the buffer sizes with mesh sizes here is arcane. It would be better for variables to know and compute their own buffer sizes and just aggregate them.

Reply:
:+1: I added a comment into this line of code, so in the next step when we turn on faces, it should work out

Reply:
This buffer size stuff also probably could be made a little bit more efficient by not packing unallocated sparse variables. When this is fixed to deal with other topological elements would be a good time to also fix the sparse data packing.